release.md

```diff
@@ -0,0 +1,298 @@
+
+
+# 2. 🐇 🐇  Web3 1.0  🐇 🐇
+
+`truffle-contract` now uses Web3 1.0 under the hood.  It's nice! The error handling (especially the parameter checking) is really good. And there's lots of new work happening over there  - ENS support is being added and [EthPrize](http://ethprize.io/) is funding an API to make contract packages published to EthPM easily available as well.
+
+We've tried to minimize the number of changes required to transition an existing test suite from Web3 0.x to 1.0. Unfortunately there are some breaking changes due to differences in the way Web3 1.0 formats outputs (see chart below).
+
+The biggest change is that `truffle-contract` now returns numbers as [BN](https://github.com/indutny/bn.js/) by default instead of BigNumber. This is necessary - BigNumber doesn't have the precision required by the EVM. The number libraries share some methods (notably `.toNumber()`) but their arithmetic operations are quite different.
+
+**Important**: If you receive a number directly from web3 (e.g. *not* from a truffle-contract instance)  you'll get a **string**. Luckily there's a helpful BN conversion method in the web3 utils.
+```javascript
+const stringBalance = await web3.eth.getBalance('0xabc..');
+const bnBalance = web3.utils.toBN(stringBalance);
+```
+
+**Helpful Resources**
++  Web3 1.0's documentation. Well worth taking a look - lots of differences in their new API.
++ EthWork's [bn-chai](https://github.com/EthWorks/bn-chai) library: BN helpers for your tests. (💡 ProTip courtesy elenadimitrova)
+
+------------
+### ⚠️ Breaking Changes ⚠️
+
+| Category | V4 (Web3 0.0) |  V5 (Web3 1.0) |
+| ------ | ---- | ------- |
+| addresses (return value) | lower-case | check-summed (mixed-case) |
+| numbers  (return value)  | BigNumber | BN (configurable) |
+| tuples (return value)       | Array | Object w/ named & indexed keys |
+| `.contract` (underlying Web3 abstraction)| same as now | completely [different](https://web3js.readthedocs.io/en/1.0/web3-eth-contract.html) |
+|`.at` (TruffleContract method)| sync / then-able | async |
+
+-----------
+### 🍨 Features 🍨
+
+### Methods / `.new` have an EventEmitter interface *and*  return a promise.
+```javascript
+example
+  .setValue(123)
+  .on('transactionHash', hash => {} )
+  .on('receipt', receipt => {})
+  .on('error', error => {})
+  .on('confirmation', (num, receipt) => {} )
+  .then( receipt => {} )
+```
+
+### Events have an EventEmitter interface
+```javascript
+example
+  .ExampleEvent()
+  .on('data', event => ... etc ... )
+
+example
+  .ExampleEvent()
+  .once('data', event => ... etc ... )
+```
+### Reason strings!! Find out the reason.
+```javascript
+# In a Solidity file
+require(msg.sender == owner, 'not authorized');
+
+// In JS
+try {
+  await example.transferToSelf({from: nonOwner})
+} catch (err) {
+  assert(err.reason === 'not authorized');
+  assert(err.message.includes('not authorized');
+}
+```
```

Comment:
This can be broken up:

**Solidity**:
```solidity
require(msg.sender == owner, 'not authorized');
```

**Javascript**:
```javascript
try {
  await example.transferToSelf({from: nonOwner})
} catch (err) {
  assert(err.reason === 'not authorized');
  assert(err.message.includes('not authorized');
}
```

Maybe not bold?

Reply:
Yeah that's much better.

Reply:
Done.